### DISCUSSION TITLE
How to Handle Proxy Objects in Computed Fields When Using Prisma with Next.js Server Actions?

### DISCUSSION THREAD

**Discussion by iGmainC** [prisma/prisma#25029] [Q&A]
### Question

I'm running into an issue when using Prisma with Next.js, specifically with computed fields that I define using the .extends() feature. The computed fields return Proxy(Object) values, which is causing serialization errors when I try to return the results directly from a Next.js Server Action.

I have a model with a computed field like this:

```javascript
const prisma = new PrismaClient().$extends({
  result: {
    collectionItem: {
      rawTextLen: {
        needs: { rawText: true },
        compute(x) {
          return x.rawText ? x.rawText.length : 0;
        },
      },
    },
  },
});
```

Then, I use the following query:

```javascript
let data = await prisma.collection.findMany({
  where: { repositoryId: repoId },
  include: {
    file: {
      select: {
        id: true,
        type: true,
      }
    },
    CollectionItem: {
      select: {
        rawTextLen: true, // Computed field
      }
    },
  },
});
```
When I try to return data directly from a Next.js Server Action:

```javascript
export async function getData() {
  return data;
}
```

The Issue:

I'm getting serialization errors because rawTextLen is returned as a Proxy(Object). This makes it difficult to work with the computed fields in a server-side environment where the results need to be sent to the client.
What I've Tried:

    Workaround 1: Manually transforming the Proxy(Object) into a plain object before returning.
    Workaround 2: Using JSON.parse(JSON.stringify(data)), but this is not ideal for performance and readability.

Questions:

    Is there an official or recommended way to avoid Proxy objects when using computed fields with Prisma?
    Has anyone else encountered this issue, and how did you solve it?
    Are there any best practices for working with computed fields in Prisma when the results need to be serialized?

Additional Context:

    The computed field rawTextLen is crucial to my application, and I'm looking for the most efficient way to handle this issue.
    This issue is blocking me from directly using computed fields in a Next.js environment where serialization is required.

Any help, advice, or pointers would be greatly appreciated!

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment]
Hi @iGmainC 👋 

There is an open issue related to this question.

- https://github.com/prisma/prisma/issues/20627

There are suggested workarounds in the open issue. Please do take a look.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
